gnd/src/main/java/com/google/android/gnd/persistence/sync/LocalMutationSyncWorker.java

```diff
@@ -77,9 +78,11 @@ public Result doWork() {
   }
 
   private Completable processMutations(ImmutableList<Mutation> pendingMutations) {
+    // TODO(#178): Why is removePendingMutations() getting run on UI thread without subscribeOn()?
```

Comment:
Looks like calling subscribeOn is a requirement for methods that return Maybe or Single:

> Room provides the option of observing the data in the database and performing asynchronous queries with the help of RxJava Maybe, Single and Flowable objects.
>
> If you’re worried about threads, Room keeps you at ease and ensures that observable queries are done off the main thread. It’s up to you to decide on which thread the events are emitted downstream, by setting the Scheduler in the observeOn method.
>
> For queries that return Maybe or Single, make sure you’re calling subscribeOn with a different Scheduler than AndroidSchedulers.mainThread().

From https://medium.com/androiddevelopers/room-rxjava-acb0cd4f3757

Reply:
@gino-m FYI